src/main/java/com/stocks/project/security/SpringSecurityConfiguration.java

```diff
@@ -0,0 +1,53 @@
+package com.stocks.project.security;
+
+import com.stocks.project.model.Role;
+import com.stocks.project.security.filter.JwtAuthenticationFilter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpMethod;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+@Configuration
+@EnableWebSecurity
+public class SpringSecurityConfiguration {
+    private final JwtAuthenticationFilter jwtAuthenticationFilter;
+
+    @Autowired
+    public SpringSecurityConfiguration(JwtAuthenticationFilter jwtAuthenticationFilter) {
+        this.jwtAuthenticationFilter = jwtAuthenticationFilter;
+    }
+
+    @Bean
+    public SecurityFilterChain filterChain(HttpSecurity httpSecurity) throws Exception {
+        return httpSecurity
+                .csrf(AbstractHttpConfigurer::disable)
+                .authorizeHttpRequests(auth -> auth
+                        .requestMatchers(HttpMethod.POST,"/authentication").permitAll()
+                        .requestMatchers(HttpMethod.POST,"/register").permitAll()
+                        .requestMatchers("/stocks").permitAll()
+
+                        .requestMatchers(HttpMethod.DELETE,"/users").hasAnyRole(Role.ADMIN.name(), Role.USER.name())
```

Comment:
Why is authorization controlled here for some endpoints, but for example for GET /getAll you check with the code whether it is an admin or not? Why can’t we check right there, so as not to clog the controllers?